alacritty_terminal/src/term/mod.rs

```diff
@@ -1878,22 +1878,36 @@ impl<T: EventListener> ansi::Handler for Term<T> {
             },
             ansi::Mode::ShowCursor => self.mode.insert(TermMode::SHOW_CURSOR),
             ansi::Mode::CursorKeys => self.mode.insert(TermMode::APP_CURSOR),
+            // Mouse protocols are mutually exlusive
             ansi::Mode::ReportMouseClicks => {
                 self.mode.insert(TermMode::MOUSE_REPORT_CLICK);
+                self.mode.remove(TermMode::MOUSE_DRAG);
+                self.mode.remove(TermMode::MOUSE_MOTION);
```

Comment:
Doing something funky like `self.mode = (self.mode ^ MOUSE_MODE) | MOUSE_REPORT_CLICK` might be an option, though I find it a bit unreadable.

But I think consolidating the `remove`s still makes sense.

I also don't think we should order by name, but rather by logical order. We should first remove everything, then insert the new ones (or the other way around). I think doing remove -> insert -> remove is a bit confusing and it's easy to miss what's done with what exactly.

We also already have a `MOUSE_MODE` which matches all other modes. So in theory we can just do `self.mode.remove(TermMode::MOUSE_MODE); self.mode.insert(new_mode);`.

Reply:
I think removing `MOUSE_MODE` and then setting a correct one it the most readable solution.